SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/EntityComponentUpdateRecordTest.cpp

```diff
@@ -0,0 +1,201 @@
+
+	// WHEN
+	Storage.AddComponentUpdate(kTestEntityId, MoveTemp(TestUpdate));
+
+	// THEN
+	TestTrue(TEXT("Updates are equal to expected"), AreEquivalent(Storage.GetUpdates(), ExpectedUpdates));
+	TestTrue(TEXT("Complete updates are equal to expected"), AreEquivalent(Storage.GetCompleteUpdates(), ExpectedCompleteUpdates));
+
+	return true;
+}
+
+ENTITYCOMPONENTUPDATERECORD_TEST(GIVEN_update_record_with_update_WHEN_new_update_added_THEN_new_update_merged)
+{
+	// GIVEN
+	auto FirstUpdate = CreateTestComponentUpdate(kTestComponentId, kTestValue);
+	auto SecondUpdate = CreateTestComponentUpdate(kTestComponentId, kTestUpdateValue);
+
+	TArray<EntityComponentUpdate> ExpectedUpdates;
+	ExpectedUpdates.Push(EntityComponentUpdate{ kTestEntityId, SecondUpdate.DeepCopy() });
+	const TArray<EntityComponentCompleteUpdate> ExpectedCompleteUpdates = {};
+	const TArray<EntityComponentUpdate> ExpectedEvents = {};
+
+	EntityComponentUpdateRecord Storage;
+	Storage.AddComponentUpdate(kTestEntityId, MoveTemp(FirstUpdate));
+
+	// WHEN
+	Storage.AddComponentUpdate(kTestEntityId, MoveTemp(SecondUpdate));
+
+	// THEN
+	TestTrue(TEXT("Updates are equal to expected"), AreEquivalent(Storage.GetUpdates(), ExpectedUpdates));
+	TestTrue(TEXT("Complete updates are equal to expected"), AreEquivalent(Storage.GetCompleteUpdates(), ExpectedCompleteUpdates));
+
+	return true;
+}
+
+ENTITYCOMPONENTUPDATERECORD_TEST(GIVEN_empty_update_record_WHEN_complete_update_added_THEN_update_record_has_complete_update)
+{
+	// GIVEN
+	auto Data = CreateTestComponentData(kTestComponentId, kTestValue);
+
+	TArray<EntityComponentCompleteUpdate> ExpectedCompleteUpdates;
+	ExpectedCompleteUpdates.Push(EntityComponentCompleteUpdate{ kTestEntityId, Data.DeepCopy(), ComponentUpdate(kTestComponentId) });
+	const TArray<EntityComponentUpdate> ExpectedUpdates = {};
+	const TArray<EntityComponentUpdate> ExpectedEvents = {};
+
+	EntityComponentUpdateRecord Storage;
+
+	// WHEN
+	Storage.AddComponentDataAsUpdate(kTestEntityId, MoveTemp(Data));
+
+	// THEN
+	TestTrue(TEXT("Updates are equal to expected"), AreEquivalent(Storage.GetUpdates(), ExpectedUpdates));
+	TestTrue(TEXT("Complete updates are equal to expected"), AreEquivalent(Storage.GetCompleteUpdates(), ExpectedCompleteUpdates));
+
+	return true;
+}
+
+ENTITYCOMPONENTUPDATERECORD_TEST(GIVEN_update_record_with_update_WHEN_complete_update_added_THEN_complete_update_merged)
+{
+	// GIVEN
+	auto Update = CreateTestComponentUpdate(kTestComponentId, kTestValue);
+	AddTestEvent(&Update, kEventValue);
+	auto CompleteUpdate = CreateTestComponentData(kTestComponentId, kUpdateValue);
+
+	const TArray<EntityComponentUpdate> ExpectedUpdates = {};
+	TArray<EntityComponentCompleteUpdate> ExpectedCompleteUpdates;
+	ExpectedCompleteUpdates.Push({ kTestEntityId, CompleteUpdate.DeepCopy(), Update.DeepCopy() });
+	TArray<EntityComponentUpdate> ExpectedEvents;
+	ExpectedEvents.Push(EntityComponentUpdate{ kTestEntityId, Update.DeepCopy() });
```

Comment:
Maybe we should check this as a TestTrue?

Reply:
Do you mean instead of using a vector?

Reply:
I am finding it hard to find this in the latest code but I believe ExpectedEvents was not being used. It seems like now it is used here:

ExpectedCompleteUpdates.Push(EntityComponentCompleteUpdate{ TestEntityId, MoveTemp(ExpectedCompleteUpdate), MoveTemp(ExpectedEvent) });

Perhaps I missed it's use before. Anyway looks fine now.